### ISSUE TITLE
Syntax colour local.anyname

### ISSUE THREAD

**Issue by rikedyp** [Dyalog/ride#494]
### Describe the issue you are having

In ride, local.anyname works but the syntax leaves .anyname black (like a global) when it should be grey (like a local)
See below for input.nodate and input.tag
![image](https://user-images.githubusercontent.com/9338369/47080630-720ba980-d200-11e8-8f3c-18a916b0b5fa.png)

### Did you connect to an already running interpreter or start the interpreter from RIDE?
Started from RIDE
### How do you reproduce the issue?
Create a tradfn (or in my case custom User Command / :NameSpace
See above

### Paste the contents of Help → About (Shift+F1)
IDE:
  Version: 4.1.3366
  Platform: Linux x86_64
  Date: 2018-07-17 15:26:35 +0100
  Git commit: 0088ce492350a8d582b019adfc5d5a87efbdc00b
  Preferences:{
    "kbdLocale":"en_GB"
  }

Interpreter:
  Version: 17.0.34137
  Platform: Linux-64
  Edition: Unicode/64
  Date: Aug 31 2018 at 10:58:46

**Comment by jayfoad** [comment]
> In ride, local.anyname works but the syntax leaves .anyname black (like a global) when it should be grey (like a local)

Yeah, _maybe_ it would be good to handle simple cases like this, but note that dot syntax gets arbitrarily complicated and there's no way syntax highlighting can do the right thing in all cases:

```apl
∇ foo;a
  (a b).c ⍝ is c local or global?
∇
```

**Comment by abrudz** [comment]
Good point. Indeed, even simple cases are ambiguous:
```apl
 fn;a
 x←1 2 3
 y←10 20 30
 a←+
 c←-
 x a.c y
```
Apparently, RIDE never attempts to colour `c` as local, while the IDE always colours `c` as local if `a` is local (even if the `.` must mean inner product).

Indeed, sometimes it is impossible to know the correct colour until execution time:
```apl
 fn;a
 x←1 2 3
 y←10 20 30
 a←⎕
 c←×
 x a.c y
```
If the user enters `⎕JSON'{"c":42}'` when prompted, then `c` is local, but if the user enters `+` when prompted, then `c` is global!
